client.cpp

```diff
@@ -490,3 +491,12 @@ PHP_METHOD(VoltClient, drain)
         RETURN_FALSE;
     }
 }
+
+PHP_METHOD(VoltClient, close)
+{
+    zval *zobj = getThis();
+    voltclient_object *obj = (voltclient_object *)zend_object_store_get_object(zobj TSRMLS_CC);
+
+    voltdb::ConnectionPool::pool()->returnClient(*(obj->client));
```

Comment:
Doesn't this just return the client back to the pool instead of closing it?

To close it, you have to call the new `close()` method @akhanzode added and make sure that you remove it from the pool. Otherwise, the next call to acquire a client on a different thread will get the closed client and fail.